arduino/smartcar/smartcar.ino

```diff
@@ -189,12 +199,75 @@ void loop()
         else if (loopControl == 4)
         {
             backUltDis = backUlt.getDistance();
-        }else if (loopControl == 5)
+        }
+        else if (loopControl == 5)
+        {
+            angleBeforeUpdate = gyroscopeAngle;
+            gyroscope.update();
+        }
+        else if (loopControl == 6)
+        {
+            gyroscopeAngle = gyroscope.getHeading();
+            angleAfterUpdate = gyroscopeAngle;
+            
+        }
+        else if (loopControl == 7)
+        {
+            carSpeed = car.getSpeed();
+
+            checkAngleDegree = angleAfterUpdate - angleBeforeUpdate;//checks if angle of change past the loop
```

Comment:
Don't understand the comment here.

Reply:
Pretty much there is a problem every time the loop goes from 360 to 1 so what this does is it checks if it hit that mark and if it does it does not give any input for those couple of milliseconds